src/main/java/com/google/sps/servlets/GmailServlet.java

```diff
@@ -66,15 +74,132 @@ public void doGet(
+   * @param gmailClient Gmail service with valid google credential
+   * @return number of unread emails from last nDays days
+   * @throws IOException if an issue occurs with the Gmail Service
+   */
+  private int getUnreadEmailCountDays(int nDays, GmailClient gmailClient) throws IOException {
+    String searchQuery = GmailClient.emailQueryString(nDays, "d", true, false, "");
+    return gmailClient.listUserMessages(searchQuery).size();
+  }
+
+  /**
+   * Get the amount of unread emails from the last m hours in a user's Gmail account
+   *
+   * @param mHours number of hours to check the user's account for
+   * @param gmailClient Gmail service with valid google credential
+   * @return number of unread emails from last mHours hours
+   * @throws IOException if an issue occurs with the Gmail Service
+   */
+  private int getUnreadEmailCountHours(int mHours, GmailClient gmailClient) throws IOException {
+    String searchQuery = GmailClient.emailQueryString(mHours, "h", true, false, "");
+    return gmailClient.listUserMessages(searchQuery).size();
+  }
+
+  /**
+   * Get the amount of unread, important emails from the last n days in a user's Gmail account
+   *
+   * @param nDays number of days to check the user's account for
+   * @param gmailClient Gmail service with valid google credential
+   * @return number of unread, important emails from last nDays days
+   * @throws IOException if an issue occurs with the Gmail Service
+   */
+  private int getImportantUnreadEmailCount(int nDays, GmailClient gmailClient) throws IOException {
+    String searchQuery = GmailClient.emailQueryString(nDays, "d", true, true, "");
+    return gmailClient.listUserMessages(searchQuery).size();
+  }
+
+  /**
+   * Get the sender of the most unread emails from the last n days in a user's Gmail account
+   *
+   * @param nDays number of days to check the user's account for
+   * @param gmailClient Gmail service with valid google credential
+   * @return sender of the most unread emails from the last nDays days
+   * @throws IOException if an issue occurs with the Gmail Service
+   */
+  private String getMostFrequentSender(int nDays, GmailClient gmailClient) throws IOException {
+    // Get user's messages - if none present, return "" as there is no sender to extract
+    String searchQuery = GmailClient.emailQueryString(nDays, "d", true, false, "");
+    GmailClient.MessageFormat messageFormat = GmailClient.MessageFormat.METADATA;
+    List<Message> unreadEmails;
+    unreadEmails =
+        gmailClient.listUserMessages(searchQuery).stream()
+            .map(
+                (Message m) -> {
+                  try {
+                    return gmailClient.getUserMessage(m.getId(), messageFormat);
+                  } catch (IOException e) {
+                    throw new RuntimeException(e);
+                  }
+                })
+            .collect(Collectors.toList());
+    if (unreadEmails.isEmpty()) {
+      return "";
+    }
```

Comment:
I wonder if it would make more sense for the auxiliary methods to live in `GmailClient`

Reply:
I actually think they would do better in the GmailResponse class, since that class has to do with the actual Gmail statistics. GmailClient is more for POST/GET requests to and from the Gmail service and any methods that govern those transactions. 

I moved them over to GmailResponse - let me know what you think!